### ISSUE TITLE
[REVIEW]: radioactivedecay: A Python package for radioactive decay calculations

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3318]
**Submitting author:** @alexmalins (<a href="http://orcid.org/0000-0003-1922-4496">Alex Malins</a>)
**Repository:** https://github.com/alexmalins/radioactivedecay/
**Branch with paper.md** (empty if default branch): joss_paper
**Version:** v0.4.9
**Editor:** @kellyrowland
**Reviewers:** @munkm, @shyamd, @kellyrowland
**Archive:** 10.5281/zenodo.6334651

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/4c2a8812e12509244fe263337a17caa7/status.svg)](https://joss.theoj.org/papers/4c2a8812e12509244fe263337a17caa7)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/4c2a8812e12509244fe263337a17caa7"><img src="https://joss.theoj.org/papers/4c2a8812e12509244fe263337a17caa7/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/4c2a8812e12509244fe263337a17caa7/status.svg)](https://joss.theoj.org/papers/4c2a8812e12509244fe263337a17caa7)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/alexmalins/radioactivedecay/" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@shyamd & @munkm, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @kellyrowland know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/alexmalins/radioactivedecay/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@alexmalins) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/alexmalins/radioactivedecay/">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@alexmalins) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### Functionality

- [x] **Functionality:** Have the functional claims of the software been confirmed?

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @bjodah, @munkm it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
PDF failed to compile for issue #3318 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.12 s (305.3 files/s, 102259.1 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          14            606           1215           2037
Jupyter Notebook                 6              0           6277           1038
Markdown                         5            135              0            427
reStructuredText                11            228            341            370
DOS Batch                        1              8              1             26
make                             1              4              7              9
-------------------------------------------------------------------------------
SUM:                            38            981           7841           3907
-------------------------------------------------------------------------------


Statistical information for the repository '813eb903faa10af58744301c' was
gathered on 2021/05/28.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Alex Malins                     41          6019           2169          100.00

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Alex Malins                3859           64.1          5.4                8.50
```

**Comment by alexmalins** [comment]
@whedon generate pdf from branch joss_paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03318/joss.03318/10.21105.joss.03318.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03318/joss.03318/10.21105.joss.03318.pdf) :page_facing_up: :point_left:

**Comment by kellyrowland** [comment]
@whedon remove @bjodah as reviewer

**Comment by whedon** [comment]
OK, @bjodah is no longer a reviewer

**Comment by kellyrowland** [comment]
Hi @shyamd 👋 - would you be available and willing to review this paper for JOSS?

**Comment by whedon** [comment]
:wave: @munkm, please update us on how your review is going (this is an automated reminder).

**Comment by shyamd** [comment]
> Hi @shyamd 👋 - would you be available and willing to review this paper for JOSS?

Sounds good!

**Comment by kellyrowland** [comment]
Great, thanks so much!

**Comment by shyamd** [comment]
Hmm. I think I need a new invite to the repository. I can't edit the checkboxes and the invite URL doesn't work.

**Comment by danielskatz** [comment]
@whedon re-invite @shyamd as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@shyamd please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by kellyrowland** [comment]
Oops - good call @shyamd and thanks @danielskatz!

**Comment by alexmalins** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
PDF failed to compile for issue #3318 with the following error: 
```
 Can't find any papers to compile :-(

```

**Comment by alexmalins** [comment]
@whedon generate pdf from branch joss_paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03318/joss.03318/10.21105.joss.03318.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03318/joss.03318/10.21105.joss.03318.pdf) :page_facing_up: :point_left:

**Comment by shyamd** [comment]
@alexmalins took care of my only concern promptly in https://github.com/alexmalins/radioactivedecay/issues/31 and I think my review is finished.

**Comment by kellyrowland** [comment]
Great, thanks so much @shyamd .

Hi @munkm - friendly reminder about this, feel free to let me know if you have any questions or concerns.

**Comment by kellyrowland** [comment]
@whedon re-invite @munkm as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@munkm please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by munkm** [comment]
@kellyrowland i've been reviewing this package and I have a question about the LICENSE file. The main license is using the MIT language, but then there are supplements to the license for the nuclear data ICRP-07 that amend it. The additions to the license should allow the package to redistribute the data for educational, not for profit, or research purposes but I am unsure if this is an issue because it does seem to fall outside of the other license texts? What do you think?

**Comment by kellyrowland** [comment]
Good catch, @munkm .

I think that the license text should be separated out into two files: one file with the MIT license text for this software package, and a second, separate license file which includes the wording for the data libraries and their respective (more restrictive) license texts. That should hopefully help differentiate the licenses.

Does that make sense? Does that work for you @alexmalins ?

**Comment by alexmalins** [comment]
@munkm & @kellyrowland

Thanks for your review and the suggestions.

Honestly, when I created the license file I wasn't sure of the best way to do it. Splitting the license file into separate files for the code (MIT) and the default data it uses (ICRP-07) works for me. Moreover, this makes sense as the code is programmed flexibly to use other datasets, and there is the potential that the default will switch to something more recent in the future.

Thanks for the issues you created in the repository, @munkm. I'll work on these and splitting the license files in the coming days.

**Comment by alexmalins** [comment]
Progress update: I've committed changes incorporating your suggestions @munkm and @kellyrowland . These are currently visible on the [devel](https://github.com/alexmalins/radioactivedecay/tree/devel) and [joss_paper](https://github.com/alexmalins/radioactivedecay/tree/joss_paper) branches of the repository.

There are two outstanding comments from the review in this [issue](https://github.com/alexmalins/radioactivedecay/issues/37). The main one is item 4, where @munkm asked for the inclusion of a statement about code only dealing with activities. Coincidentally, @lemointm worked on improvements this summer so the code now supports masses, moles and number of atoms of radioactive materials. These will be incorporated in the next release, which is basically ready to go live. I was wondering if it would be allowed to update the paper along these lines and add Thom Lemoine as a co-author?

**Comment by kellyrowland** [comment]
@alexmalins That sounds great; it's very common that software is updated in the review process and we should certainly give credit where it's due by way of adding another author.

**Comment by alexmalins** [comment]
@kellyrowland ok thanks - that's great!
We will get on with cutting the new release and updating the paper, then come back when it is done. Timescale should be a few days to a week.

**Comment by alexmalins** [comment]
@whedon generate pdf from branch joss_paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03318/joss.03318/10.21105.joss.03318.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03318/joss.03318/10.21105.joss.03318.pdf) :page_facing_up: :point_left:

**Comment by alexmalins** [comment]
Ok, everything is now ready. The paper is updated and consistent with the latest release of radioactivedecay (v0.4.2), as are the comparisons with PyNE and Radiological Toolbox. Thom Lemoine is added as a co-author to the paper. Everything that was outstanding in @munkm's review comments ([issue 37](https://github.com/alexmalins/radioactivedecay/issues/37) is closed off.

Do you know what happens with the version number of record (currently v0.3.2, written in first post in this thread by @whedon)? Is it possible to update that to v0.4.2?

**Comment by kellyrowland** [comment]
Once the review is complete, I'll have the JOSS bot update the version number so that it's up-to-date in the paper. Doing that is part of the check-out process for the editors since it's often the case that software is updated through the paper review.

**Comment by kellyrowland** [comment]
Hi @munkm 👋 friendly reminder to check in on this when you're able to.

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03318/joss.03318/10.21105.joss.03318.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03318/joss.03318/10.21105.joss.03318.pdf) :page_facing_up: :point_left:

**Comment by alexmalins** [comment]
@kellyrowland @munkm Thanks for your continued efforts on this.

We moved the project into its own organization this week and the hosting of the docs to GitHub Pages. The edits to the paper just now were to bring the URLs up to date.

**Comment by kellyrowland** [comment]
Hi @munkm 👋 friendly reminder to check in on this when you can.

**Comment by kellyrowland** [comment]
Hi @munkm 👋 friendly reminder to check in here when you can.

**Comment by kellyrowland** [comment]
I'm back in office now, so dropping in with a reminder about this @munkm 👋

**Comment by alexmalins** [comment]
@whedon generate pdf from branch joss_paper

**Comment by whedon** [comment]
```
Attempting PDF compilation from custom branch joss_paper. Reticulating splines etc...
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03318/joss.03318/10.21105.joss.03318.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03318/joss.03318/10.21105.joss.03318.pdf) :page_facing_up: :point_left:

**Comment by alexmalins** [comment]
Hi @kellyrowland @munkm,
This change was just another minor update to URLs in the paper: we moved the [dataset creation](https://github.com/radioactivedecay/datasets) and [code comparison](https://github.com/radioactivedecay/comparisons) notebooks into their own repos.

**Comment by kellyrowland** [comment]
Hi @munkm - just dropping a note here to check in on this when you can. 👋

**Comment by kellyrowland** [comment]
Hi @munkm 👋 friendly reminder to check in here when you can.

**Comment by kellyrowland** [comment]
I'm going to be finishing out the review for this paper; I have taken the remaining items from @munkm and assigned them to myself.

**Comment by kellyrowland** [comment]
@alexmalins this all looks great. 🙂 I would have loved to have a tool like this to check my own decay calculations.

**Comment by kellyrowland** [comment]
@editorialbot set v0.4.9 as version

**Comment by kellyrowland** [comment]
@editorialbot generate pdf

**Comment by kellyrowland** [comment]
@editorialbot set joss_paper as branch

**Comment by kellyrowland** [comment]
@editorialbot generate pdf

**Comment by kellyrowland** [comment]
@editorialbot check references

**Comment by kellyrowland** [comment]
@alexmalins I'm ready to recommend this for acceptance. Please create an archive of the v0.4.9 tagged release and archive it (on Zenodo, figshare, or elsewhere) and post the DOI here.

**Comment by alexmalins** [comment]
Thank you @kellyrowland

The Zenodo DOI is: [10.5281/zenodo.6334651](https://doi.org/10.5281/zenodo.6334651)

**Comment by kellyrowland** [comment]
@editorialbot set 10.5281/zenodo.6334651 as archive

**Comment by kellyrowland** [comment]
@editorialbot recommend-accept

**Comment by kellyrowland** [comment]
@editorialbot generate pdf

**Comment by kellyrowland** [comment]
@editorialbot generate pdf

**Comment by kellyrowland** [comment]
Hi @alexmalins there's a bit of inconsistency with the editoralbot behavior that the JOSS team is working to resolve; the `recommend-acceptance` command is currently failing because of some warnings. The intended behavior is that the `generate pdf` command should also be failing, so we don't get to this mismatched point.

The repeated warnings in the acceptance pipeline look to be all of the form:
```Warning:  Could not convert TeX math \mathbf{N}(t) = C e^{\varLambda_{d} t} C^{-1} \mathbf{N}(0).
  \label{eq:solution}, rendering as TeX:
  ) = C e^{\varLambda_{d} t} C^{-1} \mathb
                     ^
  unexpected control sequence \varLambda
  ``` 

I'm not seeing `varLambda` listed in http://tug.ctan.org/info/symbols/comprehensive/symbols-a4.pdf ; maybe `upLambda` (see Table 191) would be a sufficient replacement?

**Comment by xuanxu** [comment]
@alexmalins That `\varLambda` error is a problem in the [TeXMath package](https://github.com/jgm/texmath) used to parse math. The underlying issue is it's no longer enough for math to show up correctly in PDF, but that it must also be parseable by pandoc.

There's now a PR to fix `\varLambda` support in TeXMath (https://github.com/jgm/texmath/pull/190). It's been merged, so I guess the problem will be solved in the next release. 
For the time being, the following could be used as a workaround (best placed close to the top of the Markdown file): 
`\def\varLambda{\mathit{\Lambda}}`

**Comment by alexmalins** [comment]
@whedon generate pdf from branch joss_paper

**Comment by alexmalins** [comment]
@editorialbot generate pdf from branch joss_paper

**Comment by alexmalins** [comment]
@editorialbot generate pdf

**Comment by kthyng** [comment]
@alexmalins I'm taking over to finish the publishing process. Everything looks good except — in the references there are asterisks at the end of the title for several, for example Huang et al. Is there a reason for them to be there?

And hey I went to Whitman too @lemointm!

**Comment by alexmalins** [comment]
Hi @kthyng - thanks for checking this paper! Those asterisks came in the .bib files from the journal's website. They needn't be there, so this last commit removed them.

**Comment by alexmalins** [comment]
@editorialbot generate pdf

**Comment by kthyng** [comment]
@editorialbot accept

**Comment by kthyng** [comment]
Congratulations on your new paper @alexmalins!! Many thanks to editor @kellyrowland and reviewers @munkm, @shyamd, and also @kellyrowland for your time, hard work, and expertise!
